test/integration_test.py

```diff
@@ -283,11 +283,23 @@ def test_snapshot_listing(self):
         """
         Test with a small page size to be sure paging works
         """
+        snapshots = self._public_tdr_client.snapshot_names_by_id()
         page_size = 5
+        if len(snapshots) == 1:
+            log.warning('Public service account only has access to 1 snapshot')
+            page_size = 1
+        elif len(snapshots) < page_size:
+            page_size = len(snapshots) - 1
+        assert page_size >= 0
```

Comment:
1. I think there should be a branch here to handle the eventuality of receiving _zero_ snapshots, which currently would trigger this assertion.
2. I don't think this should be an `assert` due to point 1 (it depends on the external TDR response and we reserve `assert` for catching logic errors in our own code)
3. I think the assert condition is too lenient. I don't think the test should continue with a page size of zero.

Reply:
I addressed 2/3. But I'm not sure of the usefulness of a branch that would read something like.
```
elif len(snapshots) == 0:
    require(page_size > 0)
```
Seems like an extra line.

Reply:
Makes sense. My first and second point kind of cancel each other out.